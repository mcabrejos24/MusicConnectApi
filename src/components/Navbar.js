import React from 'react';

export default function Navbar() {
    
    return(
        <header className="w-full bg-black fixed">
            <div class="h-20 flex  xl:px-36 lg:px-24 md:px-20 justify-center md:justify-between">
                <a href="/" className="self-center font-bold text-white text-xl">
                    Playlist Connect
                </a>
                <div className="hidden md:flex md:flex-row md:self-center md:font-light">
                    <a className="hover:underline text-white text-lg pr-4" href="#P3L">Learn more</a>
                    <a className="hover:underline text-white text-lg px-4" href="#P3L">Donate</a>
                    <a className="hover:underline text-white text-lg pl-4" href="#P3L">Contact Us</a>
                </div>
            </div>
          </header>
    );
} 
