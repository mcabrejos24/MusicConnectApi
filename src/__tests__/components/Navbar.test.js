import React from "react";
import { shallow } from "enzyme";
import Navbar from '../../components/Navbar';
import toJson from "enzyme-to-json";

describe("renders Navbar", () => {
    let page;
    beforeEach(() => {
        page = shallow(<Navbar />);
    });
    it("Navbar loads", () => {
        page;
    });
    it("props class name matches", () => {
        expect(page.props().className).toEqual("navbar");
    });
    it("matches snapshot", () => {
        expect(toJson(page)).toMatchSnapshot();
    });

    it("can open mobile navigation bar", () => {
        page.find(".navbar-wrapper").find(".mobile-nav").simulate("click");
        expect(page.find(".navbar-wrapper").find(".site-options").hasClass("show-site-options")).toEqual(true);
        expect(toJson(page)).toMatchSnapshot();
    });
    it("changes mobile navbar icon to cancel", () => {
        page.find(".navbar-wrapper").find(".mobile-nav").simulate("click");
        expect(page.find(".navbar-wrapper").find(".mobile-nav").find(".menu").hasClass("hide-menu-icon")).toEqual(true);
        expect(page.find(".navbar-wrapper").find(".mobile-nav").find(".cancel").hasClass("show-cancel-icon")).toEqual(true);
    });
    it("can close mobile navigation bar", () => {
        page.find(".navbar-wrapper").find(".mobile-nav").simulate("click");
        page.find(".navbar-wrapper").find(".mobile-nav").simulate("click");
        expect(page.find(".navbar-wrapper").find(".site-options").hasClass("show-site-options")).toEqual(false);
        expect(toJson(page)).toMatchSnapshot();
    });
    it("changes mobile navbar icon to menu", () => {
        page.find(".navbar-wrapper").find(".mobile-nav").simulate("click");
        page.find(".navbar-wrapper").find(".mobile-nav").simulate("click");
        expect(page.find(".navbar-wrapper").find(".mobile-nav").find(".menu").hasClass("hide-menu-icon")).toEqual(false);
        expect(page.find(".navbar-wrapper").find(".mobile-nav").find(".cancel").hasClass("show-cancel-icon")).toEqual(false);
    });
});
