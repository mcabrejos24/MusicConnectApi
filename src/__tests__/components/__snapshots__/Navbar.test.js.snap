// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar can close mobile navigation bar 1`] = `
<header
  className="navbar"
>
  <div
    className="navbar-wrapper"
  >
    <button
      className="mobile-nav"
      onClick={[Function]}
    >
      <img
        alt="menu icon"
        className="mobile-nav-icon menu "
        src="menu.svg"
      />
      <img
        alt="cancel icon"
        className="mobile-nav-icon cancel "
        src="cancel.svg"
      />
    </button>
    <a
      className="site-name"
      href="/"
    >
      Playlist Pair
    </a>
    <div
      className="site-options "
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about-us"
      >
        About Us
      </a>
      <a
        href="/donate"
      >
        Donate
      </a>
      <a
        href="/contact-us"
      >
        Contact Us
      </a>
    </div>
  </div>
</header>
`;

exports[`renders Navbar can open mobile navigation bar 1`] = `
<header
  className="navbar"
>
  <div
    className="navbar-wrapper"
  >
    <button
      className="mobile-nav"
      onClick={[Function]}
    >
      <img
        alt="menu icon"
        className="mobile-nav-icon menu hide-menu-icon"
        src="menu.svg"
      />
      <img
        alt="cancel icon"
        className="mobile-nav-icon cancel show-cancel-icon"
        src="cancel.svg"
      />
    </button>
    <a
      className="site-name"
      href="/"
    >
      Playlist Pair
    </a>
    <div
      className="site-options show-site-options"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about-us"
      >
        About Us
      </a>
      <a
        href="/donate"
      >
        Donate
      </a>
      <a
        href="/contact-us"
      >
        Contact Us
      </a>
    </div>
  </div>
</header>
`;

exports[`renders Navbar matches snapshot 1`] = `
<header
  className="navbar"
>
  <div
    className="navbar-wrapper"
  >
    <button
      className="mobile-nav"
      onClick={[Function]}
    >
      <img
        alt="menu icon"
        className="mobile-nav-icon menu "
        src="menu.svg"
      />
      <img
        alt="cancel icon"
        className="mobile-nav-icon cancel "
        src="cancel.svg"
      />
    </button>
    <a
      className="site-name"
      href="/"
    >
      Playlist Pair
    </a>
    <div
      className="site-options "
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about-us"
      >
        About Us
      </a>
      <a
        href="/donate"
      >
        Donate
      </a>
      <a
        href="/contact-us"
      >
        Contact Us
      </a>
    </div>
  </div>
</header>
`;
